tests/playbooks/e2e.yaml

```diff
@@ -41,96 +41,56 @@
       retries: 18
 
     - name: Create a VM with the CDI PVC that was created
-      k8s:
-       state: present
-       name: '{{ vm_name }}'
-       namespace: default
-       inline:
-         apiVersion: kubevirt.io/v1alpha2
-         kind: VirtualMachine
-         metadata:
-           name: '{{ vm_name }}'
-           namespace: default
-           labels:
-             kubevirt.io/vm: test
-         spec:
-           running: false
-           template:
-             metadata:
-               labels:
-                 kubevirt-vm: cirros
-                 special: cirros-e2e-key
-             spec:
-               domain:
-                 resources:
-                   requests:
-                     memory: 512Mi
-                 cpu:
-                   cores: 1
-                 devices:
-                   disks:
-                   - volumeName: disk0-pvc
-                     name: mydisk
-                     disk:
-                       bus: virtio
-                   - name: cloudinitdisk
-                     volumeName: cloudinitvolume
-                     cdrom:
-                       bus: virtio
-               volumes:
-               - name: disk0-pvc
-                 persistentVolumeClaim:
-                   claimName: '{{ pvc_name }}'
-               - name: cloudinitvolume
-                 cloudInitNoCloud:
-                   userData: |
-                     #cloud-config
-                     password: "gocubsgo"
-                     chpasswd: ( expire: False )
+      kubevirt_vm:
+        state: present
+        name: "{{ vm_name }}"
+        namespace: default
+        memory: 64Mi
+        inline:
+          spec:
+            template:
+              spec:
+                domain:
+                  cpu:
+                    cores: 1     
+        disks:
+        - name: rootdisk
+          volume:
+            persistentVolumeClaim:
+              claimName: "{{ pvc_name }}"
+          disk:
+            bus: virtio
 
     - name: Change VM status to running
       kubevirt_vm:
         state: running
-        name: '{{ vm_name }}'
+        name: "{{ vm_name }}"
         namespace: default
 
     - name: Wait for VM to be ready
       k8s_facts:
         name: "{{ vm_name }}"
         namespace: default
         kind: VirtualMachine
-        api_version: kubevirt.io/v1alpha2
+        api_version: kubevirt.io/v1alpha3
```

Comment:
Why do we need `api_version` attribute here?

Reply:
Because `k8s_facts` needs it (or can it be avoided?)
https://github.com/ansible/ansible/blob/f49469f7bf42297ec47de16e7a5e851fbf1f442b/lib/ansible/modules/clustering/k8s/k8s_facts.py#L161

I tried to run without specifying the `api_version` and it just returns blank array, which is wrong

Reply:
OK, this means we need to simplify facts module as well. I will open a card to track it.